packages/jest-diff/src/__tests__/diff-test.js

```diff
@@ -68,6 +68,12 @@ test('oneline strings', () => {
   expect(stripAnsi(diff('123456789', '234567890'))).toBe(null);
 });
 
+test('falls back to not call toJSON if objects look identical', () => {
```

Comment:
probably worth adding a test case where two objects have the same internal structure as well